baybe/kernels/base.py

```diff
@@ -19,20 +18,32 @@
 class Kernel(ABC, SerialMixin):
     """Abstract base class for all kernels."""
 
-    lengthscale_prior: Optional[Prior] = field(default=None, kw_only=True)
-    """An optional prior on the kernel lengthscale."""
-
     def to_gpytorch(self, *args, **kwargs):
         """Create the gpytorch representation of the kernel."""
         import gpytorch.kernels
 
         kernel_cls = getattr(gpytorch.kernels, self.__class__.__name__)
         fields_dict = filter_attributes(object=self, callable_=kernel_cls.__init__)
 
-        # If a lengthscale prior was chosen, we manually add it to the dictionary
-        if self.lengthscale_prior is not None:
-            fields_dict["lengthscale_prior"] = self.lengthscale_prior.to_gpytorch()
-
+        # If there are any priors in the attributes, we transform them using our
```

Comment:
Your comments (here and everywhere below) pretty much describe 1:1 the actions that happen in the code (i.e. I can **see** that `to_gpytorch` is called and no args and kwargs are passed – no need to repeat that) but they don't make it any clearer WHY this is the case. That is, WHY don't we need to pass (kw)args here but *do* need to pass them below?

In general: either a step is trivial, and then a comment merely has the function to "group" a set of lines into one higher-level semantic operation and describe the meaning of that operation in the form of brief heading (in my opinion), or the a comment should justify a certain decision and provide details / context. Not so much the case here...

Reply:
I have for now deleted the comments since this part will probably undergo some changes due to what we discussed anyway.